tftrt/examples/cpp/image-classification/main.cc

```diff
@@ -0,0 +1,168 @@
+/* Copyright 2021 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include <google/protobuf/map.h>
+#include <iostream>
+#include <stdint.h>
+#include <string>
+#include <vector>
+
+#include "mnist.h"
+#include "tensorflow/cc/saved_model/loader.h"
+#include "tensorflow/compiler/tf2tensorrt/trt_convert.h"
+#include "tensorflow/core/framework/tensor.pb.h"
+#include "tensorflow/core/lib/core/status.h"
+#include "tensorflow/core/public/session.h"
+
+tensorflow::Status LoadModel(const std::string &export_dir,
+                             tensorflow::SavedModelBundle *bundle) {
+
+  tensorflow::RunOptions run_options;
+  tensorflow::Status status = tensorflow::LoadSavedModel(
+      tensorflow::SessionOptions(), run_options, export_dir, {"serve"}, bundle);
+
+  if (!status.ok()) {
+    std::cerr << "Error loading saved model " << status << std::endl;
+    return status;
+  }
+
+  // Print the signature defs.
+  // This keras model should have an input named
+  // "serving_default_flatten_input", and an output named
+  // "StatefulPartitionedCall".
+  auto signature_map = bundle->GetSignatures();
+  for (const auto &name_and_signature_def : signature_map) {
+    const auto &name = name_and_signature_def.first;
+    const auto &signature_def = name_and_signature_def.second;
+    std::cerr << "Name: " << name << std::endl;
+    std::cerr << "SignatureDef: " << signature_def.DebugString() << std::endl;
+  }
+
+  return tensorflow::Status::OK();
+}
+
+// We prepare to input sets, with batch size 1 and batch size 4.
+tensorflow::Status
+LoadInputs(std::vector<std::vector<tensorflow::Tensor>> *inputs) {
+  // Load the MNIST images from the given path.
+  std::vector<mnist::MNISTImage> images;
```

Comment:
nit: It seems like the vector of `mnist::MNistImage` should be owned by the MNist loader class